File: loops/notes.py
# Douglas london, Loops Notes Python


# What is a loop? 
    # section of code that repeats multiple times
# What are the two types of loops?
    #for loop
nums = [12,3,66,7,3,3,2]

for num in nums:
    print(num)

    # While loop
x = 0

while x < 10:
    print(x)
    x+= 1
# What is iteration
    # That specific instance of the loop
    # iteration is the amount of times you are looping through
# What are lists? 
    # A variable that holds multiple values
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
siblings = ["Sara", "Bennett", "Rosalie", "Brielle"]
print(nums)
print(siblings[3])

siblings[2] = "Douglas"
siblings.pop(2)
siblings.insert(1, "Douglas")
# How do you make lists in python? 
    # use brackets, add items with correct data types, comma between each item
# How do you make for loops in python? 
for sibling in siblings:
    print(sibling)

for x in range(0,999999, 1):
    print (x)
# How do you make while loops in python?
import random

x = 1 # variable to keep count of iteration
goose = random.randint(1,20)


while x <= 20:
    if x == goose:
        print("Goose!")
        break
    else:
        print("Duck")
        x==1

#continue moves to next itteration 
    

File: loops/notes.c
// Douglas London, loops notes C
#include <stdio.h>

int main(void){
  

    
    

//What is a loop? 
    //loops certain peices of code
//What are the two types of loops?
    //while loops
    int start = 0;
    while(start <5){
        printf("Hello!\n");
        start++;
    }
    

    //for loop
    int q; 
    for(q=0;q<5;q++){
        printf("%d\n", q);
    }
//What is iteration
    // repeating something with minor changes ecery time
//What are arrays(list? 
    // a variable holding multiple values
//How do you make arrays(lists) in C? 
    // array items must all be same data point
int grades[] = {88, 97, 100, 70, 72, 99, 61};
    // 1. set data type 2. AFTER naming put brackets and write the lengthe of the list 3. list must bbe sourounded by curly brackets 4. must use commas between each item of the list
    // print a single item from list
printf("CSP grade: %d\n", grades[2]);
    // change an item
grades[2] = 95;
printf("CSP grade: %d\n", grades[2]);
    //size of list in bytes
int size = sizeof(grades);
 // lenght in list of items
 int length = sizeof(grades)/(grades[0]);
printf("%d\n", length);

    //array with strings
    // first [] sets lenght of list
    //second brackets sets lenght of each string
char movies[][20] = {"Cars", "Treasure Planet", "An American Tale", "Marley and Me", "The Avengers"}; 
printf("The movie is %s\n", movies[1]);
int mlenght = sizeof(movies)/sizeof(movies[0]);
//How do you make for loops in C?
    //set the itterator, keeps track of times throiugh the loop (best to set as x or y)
int x;
    // in parens (starting point; ending point; what we count by)
for(x=0;x<=10;x+=2){
    printf("%d\n", x);
}
int m;
for(m=0;m<mlenght;m++){
    printf("%s\n", movies[m]);
}

//How do you make while loops in C?
    int w = 0;

    while(w<=100){
        printf("%d\n", w);
        w += 10;
    }


    return 0;
}

